### DISCUSSION TITLE
HTTP Basic auth with static route

### DISCUSSION THREAD

**Discussion by nebnes** [vercel/next.js#24560] [Help] [answered]
Hello, 

I am trying to find a solution to protect a route (static) with HTTP Basic auth. 
I have protected the route with nginx. But when I click on the link a soft loading is execute and the page is load without pass trough the HTTP auth. I have try also to use a simple "a" tag instead of "next/Link" but when I export the site the html is not generated for this page. 

Have you an idea ? 
Thanks in advance

**Comment by akellbl4** [comment, marked as answer]
Hey,
Have you tried to protect the request to JSON as well? I mean when you go to a page via `next/link` it requests JSON for rendering and that JSON has the name of the page. You can try to add something like `_next/.*/page.json$` where `page` is the name of the protected page.

**Reply by nebnes** [reply]
Thank you @akellbl4 

I have protected the following resources for a "$page" with Nginx:
`^/(assets/$page|$page|_next/(.*)/$page.json)`
and I also replace the <next/Link> lead to this page by a tag to avoid automatic prefetch (which triggers the http auth prompt).
